src/controllers/dockController.ts

```diff
@@ -17,35 +16,37 @@ export const registerDock = async (
   res: Response
 ): Promise<unknown> => {
   const errors = validationResult(req);
-  let ReferenceId = generateRefrenceId();
   const { StandId } = req.body;
   if (!errors.isEmpty()) {
     return res.status(400).json({ errors: errors.array() });
   }
-  const exists = await prisma.cycleDock.findFirst({
+  const exists = await prisma.cycleStand.findFirst({
     where: {
-      ReferenceId: ReferenceId
+      Id: StandId
     },
     select: {
-      ReferenceId: true
+      Id: true
     }
   });
-  if (exists?.ReferenceId && exists.ReferenceId) {
-    ReferenceId = generateRefrenceId();
+  if (!exists?.Id || !exists.Id) {
+    const code = 409;
+    const file = 'dock';
+    res.render('components/response.ejs', { code: code, file: file });
```

Comment:
Instead of sending flags like file, error message itself can be sent from the server and displayed in the frontend.